plugin/code_actions.py

```diff
@@ -302,7 +302,12 @@ def combine_commands(self) -> 'List[Tuple[str, str, CodeActionOrCommand]]':
 
     def show_popup_menu(self) -> None:
         if len(self.commands) > 0:
-            self.view.show_popup_menu([command[1] for command in self.commands], self.handle_select)
+            items = [command[1] for command in self.commands]
+            win = self.view.window()
+            if win:
+                win.show_quick_panel(items, self.handle_select, placeholder="Code action")
+            else:
```

Comment:
I think that there is no point in having this else branch. 
It can be removed.
Pretty certain that `if win:` will always be true.

Reply:
I needed to add the `if` otherwise mypy won't be happy as `window` is an `Optional[Window]`, so I also put the `else` in there in case `window` is indeed `None` that way we sure we'll have our precious code actions!

Reply:
If a window doesn't exist, can the view exist?(because the view belongs to a window) 

I still think that there is no need to have both `window.show_quick_panel`(will work in 99% of the cases) and `view.show_popup_menu`(as fallback)
```diff
            if window:
                window.show_quick_panel(items, self.handle_select, placeholder="Code action")
-           else:
-                 self.view.show_popup_menu(items, self.handle_select)
```

The only situation where this will hit the else branch is when `window_id == 0` which is almost never. (I forgot the case where the window will be none. Maybe when we set a timeout and close the window in the mean time)
```python
# copied from sublime.py
class View:
    def window(self):
        window_id = sublime_api.view_window(self.view_id)
        if window_id == 0:
            return None
        else:
            return Window(window_id)
```